lib/application/widgets/theme/tags_field.dart

```diff
@@ -0,0 +1,273 @@
+          ]
+        ],
+      ).withSymmetricalPadding(context, vertical: Spacing.small, horizontal: Spacing.medium),
+    );
+
+    final helperText = Text(
+      errorText != null ? errorText! : strings.tagsAmount(tags.value.length, maxTags),
+      style: textTheme.caption?.copyWith(
+        color: errorText != null ? theme.destructiveSwatch : theme.neutralSwatch.shade400,
+      ),
+    );
+
+    return GestureDetector(
+      onTap: () {
+        fieldFocus.requestFocus();
+        isFocused.value = true;
+      },
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          tagsFieldContainer,
+          context.verticalBox(Spacing.xxxSmall),
+          helperText.withOnlyPadding(context, left: Spacing.small)
+        ],
+      ),
+    );
+  }
+}
+
+/// The custom tagged TextField.
+///
+/// Wraps the already chosen tags with an input [TextField] used to add new tags.
+class _TagsTextField extends HookConsumerWidget {
+  const _TagsTextField({required this.controller, this.fieldController, this.focus, this.onSubmitted});
+
+  final TagsController controller;
+  final TextEditingController? fieldController;
+
+  /// Controls the current field focus.
+  final FocusNode? focus;
+
+  /// Called when the user indicates they are done inputting `tag`.
+  ///
+  /// Usually fired when the user press the "done" keyboard button.
+  final Function(String tag)? onSubmitted;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+    final textTheme = Theme.of(context).textTheme;
+
+    final tags = useState(controller.tags);
+    void onTagTap(String tag) {
+      controller.tags = controller.tags..remove(tag);
+      tags.value = controller.tags;
+    }
+
+    final tagsWidgets = controller.tags.map((tag) => _SelectedTag(tag: tag, onTap: onTagTap));
+
+    final textField = TextField(
+      focusNode: focus,
+      controller: fieldController,
+      cursorColor: theme.secondarySwatch,
+      autocorrect: false,
+      enableSuggestions: false,
+      decoration: InputDecoration(
+        isCollapsed: true,
+        contentPadding: EdgeInsets.zero,
+        fillColor: Colors.transparent,
+        hintText: strings.tagsHint,
+        hintStyle: textTheme.bodyText2?.copyWith(color: theme.neutralSwatch.shade600),
+      ),
+      textAlignVertical: TextAlignVertical.center,
+      style: textTheme.bodyText2,
+      keyboardType: TextInputType.text,
+      textInputAction: TextInputAction.done,
+      inputFormatters: [_TagFieldFormatter()],
+      onSubmitted: (term) {
+        fieldController?.clear();
+        onSubmitted?.call(term);
+        focus?.requestFocus();
+      },
+    );
+
+    return Wrap(
+      spacing: context.rawSpacing(Spacing.small),
+      runSpacing: context.rawSpacing(Spacing.small),
+      children: [
+        ...tagsWidgets,
+        IntrinsicWidth(child: textField),
```

Comment:
IntrinsicWidth is really expensive on the UI, don't we have any alternative for this here?

Reply:
Unfortunately not. This is one of those scenarios where the `Intrinsic` Widgets must be used, otherwise it would be possible to place the `TextField` used to input the tags right next to the tags itself, because the `TextField` would stretch to fit the entire width